enhancements/ingress/wildcard-routing.md

```diff
@@ -0,0 +1,189 @@
+
+## Summary
+
+See the [section][#wildcard-routing-policy]
+
+## Motivation
+
+See the [section][#wildcard-routing-policy]
+
+### Goals
+
+1. Define an API field that allows a cluster administrator to enable wildcard routing for an ingress controller.
+
+### Non-Goals
+
+1. Redesign how wildcard routing is implemented in [OpenShift Router](https://github.com/openshift/router).
+2. Introduce a separate API for managing wildcard routing.
+
+## Proposal
+
+Introduce a `WildcardRouting` field to the [IngressController](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go)
+resource that allows administrators to enable or disable wildcard routing.
+
+### User Stories
+
+### Implementation Details/Notes/Constraints [optional]
+
+The [Ingress Operator](https://github.com/openshift/cluster-ingress-operator) will use `WildcardRouting` to configure
+the environment variable `ROUTER_ALLOW_WILDCARD_ROUTES` in the [OpenShift Router](https://github.com/openshift/router).
+
+The default behavior should be to disallow wildcard routing, which is backwards compatible with existing
+`IngressController` resources.
+
+### Risks and Mitigations
+
+Addition of the new field requires no migration, as the default is not changing.
+
+## Design Details
+
+### New API Field
+
+`WildcardRouting` enum definition.
+
+```go
+// (Existing fields omitted, only newly proposed fields are visible)
+type RouteAdmissionPolicy struct {
+  // wildcardRouting enables or disables wildcard routing support for
+  // the ingress controller. WildcardRouting works in conjunction with
+  // the WildcardPolicy of routes [1] claimed by the ingress controller.
+  //
+  // [1] https://github.com/openshift/api/blob/master/route/v1/types.go
+  //
+  // WildcardRouting can not be updated.
+  //
+  // If empty, defaults to "Disabled".
+  //
+  WildcardRouting *WildcardRoutingType `json:"wildcardRouting,omitempty"`
+}
+
+const (
+  // WildcardRoutingDisabled indicates only routes with a wildcard policy
+  // of "None" are admitted by the ingress controller.
+  WildcardRoutingDisabled WildcardRoutingType = "Disabled"
+
+  // WildcardRoutingEnabled indicates routes with any wildcard policy are
+  // admitted by the ingress controller.
+  WildcardRoutingEnabled WildcardRoutingType = "Enabled"
+)
+
+// (Existing type and field, included for completeness)
+type IngressControllerSpec struct {
+  // routeAdmission defines a policy for handling new route claims (for example,
+  // to allow or deny claims across namespaces).
+  //
+  // If empty, defaults will be applied. See specific routeAdmission fields
+  // for details about their defaults.
+  //
+  // +optional
+  RouteAdmission *RouteAdmissionPolicy `json:"routeAdmission,omitempty"`
+}
+
+// (Existing type and field, included for completeness)
+type IngressControllerStatus struct {
+  // routeAdmission is the route admission policy that is in effect.
+  RouteAdmission *RouteAdmissionPolicy
```

Comment:
`IngressControllerStatus` does not have this field.

Reply:
@Miciah https://github.com/openshift/enhancements/pull/262 removes route admission status from the EP.